File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/DocumentArchives/DocumentSeriesItemRole.cs
using System;
using VecompSoftware.DocSuiteWeb.Entity.Commons;

namespace VecompSoftware.DocSuiteWeb.Entity.DocumentArchives
{
    public class DocumentSeriesItemRole : DSWBaseEntity
    {
        #region [ Constructor ]

        public DocumentSeriesItemRole() : this(Guid.NewGuid()) { }
        public DocumentSeriesItemRole(Guid uniqueId)
            : base(uniqueId)
        {

        }
        #endregion

        #region [ Properties ]        
        public DocumentSeriesItemRoleLinkType LinkType { get; set; }
        #endregion

        #region [ Navigation Properties ]
        public virtual DocumentSeriesItem DocumentSeriesItem { get; set; }
        public virtual Role Role { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/DocumentSeries/domain/DocumentSeriesItemRole.vb

<Serializable()> _
Public Class DocumentSeriesItemRole
    Inherits DomainObject(Of Int32)

    Public Overridable Property UniqueIdDocumentSeriesItem As Guid

    Public Overridable Property Item As DocumentSeriesItem
    Public Overridable Property Role As Role
    Public Overridable Property LinkType As DocumentSeriesItemRoleLinkType

End Class